Add Automated Script to Reset Subscription Usage Count Every 30 Days

DeltaDynamo opened the pull request:
### Summary

This pull request introduces a new script that is scheduled to run automatically at midnight to refresh the usage count of subscriptions. 

### Description

The script performs the following actions:
- Connects to the MongoDB database and retrieves all subscriptions.
- Calculates the number of days since each subscription was created.
- Resets the usage count to zero for subscriptions after every 30 consecutive days starting from the time of creation and only if the current usage count is greater than zero.
- Updates the `updatedAt` field to reflect the reset action.

This PR resolves [Issue #355](https://github.com/TeamShiksha/logoexecutive/issues/355)

### How to Test the Changes

<!-- Describe the steps to test your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. -->

### Context (Optional)

<!-- Provide any additional context about the problem or feature here. -->

### Screenshots or Recordings (Optional)

![Screenshot 2024-08-23 234213](https://github.com/user-attachments/assets/5506a7e6-ac8e-4d7d-b83d-94e2a9ee069d)


## Checklist

<!-- To tick a checkbox, change '[ ]' to '[x]' -->

- [X] I have tested the changes locally and they work as expected.
- [ ] I have added/updated tests that cover the changes.
- [ ] I have updated the documentation to reflect the changes.
- [X] This pull request follows the project's coding guidelines.

DeltaDynamo committed:
Add refreshUsageCount script

DeltaDynamo committed:
Update refresh-usage-count.yml

DeltaDynamo committed:
Update refresh-usage-count workflow

DeltaDynamo committed:
Merge branch 'dev' of https://github.com/DeltaDynamo/logoexecutive into dev

DeltaDynamo committed:
Update refresh-usage-count.yml

DeltaDynamo committed:
Update script logic + minor workflow fix

DeltaDynamo committed:
Update workflow

DeltaDynamo committed:
Fix refreshUsageCount logic + Add timezone functionality to workflow

aps08 commented on server/scripts/refreshUsageCount.js:
@amankumarsingh77 can you confirm if we use promise all. The number of API call per second will hit the limit for the mongoDB cluster. Consider that the number of users > 500. @DeltaDynamo you might need to change the approach if it hits the limit.

aps08 replied:
Also remove the comments

DeltaDynamo replied:
Hi Anoop, you are right. I will look into it and update the code accordingly.

DeltaDynamo committed:
Optimize script logic to reduce no of API calls made to DB

DeltaDynamo committed:
Remove console logs

DeltaDynamo commented:
Hi @aps08 Following your last comment. I have made the following changes.

- I have removed individual save for records, instead of it, now using bulkWrite() to update multiple records at once. This will reduce the unnecessary API calls that was earlier being made. As a result, promise.all() was also not required so removed it as well.
- I have updated the find query to fetch only those records where usageCount > 0 because anyway we are going to update only those records where usageCount is greater than 0.

Kindly review the changes made. Let me know if you find any issues with the current approach.
